contracts/StreamManager.vy

```diff
@@ -1,259 +1,532 @@
 
+    log StreamClaimed(stream_id, msg.sender, funded_amount == claim_amount, claim_amount)
 
-@view
-def _stream_is_cancelable(creator: address, stream_id: uint256) -> bool:
-    # Creator needs to wait `MIN_STREAM_LIFE` to cancel a stream
-    return self.streams[creator][stream_id].start_time + MIN_STREAM_LIFE <= block.timestamp
+    return claim_amount
 
 
-@view
 @external
-def stream_is_cancelable(creator: address, stream_id: uint256) -> bool:
-    return self._stream_is_cancelable(creator, stream_id)
+def claim_stream(stream_id: uint256) -> uint256:
+    """
+    @dev Claim all vested tokens from Stream `stream_id` and transfer to `controller`.
+    @notice This function is unauthenticated and can be called by anyone. This can allow any number
+        of use cases such as allowing automated revenue collection and optimization, or to give the
+        gift of gas money to `controller`!
+    @param stream_id The identifier of the Stream to claim vested tokens for.
+    @return claim_amount The amount of tokens that have been transferred to `controller`.
+    """
+    return self._claim_stream(stream_id)
 
 
 @external
-def cancel_stream(
-    stream_id: uint256,
-    reason: Bytes[MAX_REASON_SIZE] = b"",
-    creator: address = msg.sender,
-) -> uint256:
-    if msg.sender == creator:
-        assert self._stream_is_cancelable(creator, stream_id)
-    else:
-        # Owner can cancel at any time
-        assert msg.sender == self.owner
+def fund_stream(stream_id: uint256, amount: uint256, min_stream_life: uint256 = 0) -> uint256:
+    """
+    @dev Add `amount` tokens worth of funding to Stream `stream_id`, to extend it's `time_left`.
+    @notice This function is unauthenticated and can be called by anyone. This can allow any
+        number of use cases such as allowing service self-payment, handling partial refunds or
+        settling disputes, or simply gifting users the gift of more time!
+    @param stream_id The identifier of the Stream to add `amount` of tokens for.
+    @param amount The total amount of tokens to add for Stream `stream_id`.
+    @return time_left The new amount of time left in Stream `stream_id`.
+    """
+    # NOTE: Anyone can fund a stream
+    token: IERC20 = self.streams[stream_id].token
+    assert self.token_is_accepted[token]  # dev: token not accepted
+    assert extcall token.transferFrom(
+        msg.sender, self, amount, default_return_value=True
+    )
+    assert block.timestamp < self.streams[stream_id].expires_at
 
-    funded_amount: uint256 = self.streams[creator][stream_id].funded_amount
-    amount_locked: uint256 = funded_amount - self._amount_unlocked(creator, stream_id)
-    assert amount_locked > 0  # NOTE: reverts if stream doesn't exist, or already cancelled
-    self.streams[creator][stream_id].funded_amount = funded_amount - amount_locked
+    # Make sure stream claims are up-to-date (ensures that stream rate math is correct)
```

Comment:
Make a note that this is an important security property because if the funding rate changes then the math for claiming will no longer be accurate.

Reply:
```suggestion
    # NOTE: Stream claims must be up-to-date to ensure that math is correct in `_amount_claimable()`
    #       This is because the stream rate may change in `_compute_stream_life()`
```